### PR TITLE
[Jetpack Social] Implement "Not now" button in post settings no connections container

### PR DESCRIPTION
**Pull request by RenanLukas**
Part of #18571

Should only be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/18869

To test:
1 - Enable JetpackSocialFeatureConfig feature flag;
2 - Make sure you have **no** sharing accounts connected;
3 - Select a SH site with Jetpack plugin or Jetpack Social plugin installed;
4 - Open post list;
5 - Select a draft or create a new one;
6 - Tap the overflow menu and select "Post settings";
7 - Scroll down to the "Social Sharing" section;
9 - The no connections UI should be shown, with the buttons `"Connect accounts"` and `"Not now"`;
10 - Tap `"Not now"`: the "Social Sharing" container should be gone;
11 - Close the screen and open it again: the container should still be gone;
12 - Add some social account using `My Site` -> `Menu` -> `Social` and open post settings again: the "Social Sharing" loaded state should be shown;
13 - Remove the social connection and open post settings again: no "Social Sharing" section should be shown, including the no connections UI.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated `EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt` and `EditPostPublishSettingsViewModelTest.kt`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 1]
```diff
@@ -162,13 +165,24 @@ class EditPostPublishSettingsViewModel @Inject constructor(
             return
         }
         siteModel?.let {
-            _showJetpackSocialContainer.value = true
-            val state = if (showNoConnections()) {
-                jetpackUiStateMapper.mapNoConnections(::onJetpackSocialConnectProfilesClick)
+            if (showNoConnections()) {
+                val shouldShowJetpackSocialNoConnections =
+                    appPrefsWrapper.getShouldShowJetpackSocialNoConnections(it.id, JetpackSocialFlow.POST_SETTINGS)
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt]: IINM the `getShouldShowJetpackSocialNoConnections` seems to be using the `SHOULD_HIDE_JETPACK_INSTALL_FULL_PLUGIN_CARD` key, that to me it seems related to another feature. Is that intended? 🤔 ...also, the function naming is about showing the no connections, while the key is about hiding something (that might be confusing I guess). Wdyt?

**Review by develric — COMMENTED** [review]
Hey @RenanLukas 👋 , I tested the PR as per test steps and all worked as described. I left a note inline with the code, let me know wdyt 🙇 

PS: on a side note for tracking purpose, I see the no connections card includes the Twitter logo but we are currently not supporting it. I know you are preparing another small PR that also fixes that, so I'm ok to have it fixed there 🙇

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt]: Good catch, thanks! I had copy + pasted this method and forgot to replace the key 😅

**Comment by RenanLukas** [comment]
Thanks for the review, @develric ! I've addressed your comment.

**Review by develric — APPROVED** [review]
Thanks for the changes 🙇 ! LGTM 👍 . Just for tracking, I see you used the remote site id instead of the local site id for composing the prefs key and I think it makes sense since connecting socials is meant for sites having a unique remote site id (that is JP connected or .com sites). Feel free to merge as soon as the CI is green :shipit:

**Pull request merged**